script.js

```diff
@@ -1,4 +1,103 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+/*
+1. There are 2 players and players take turns.
+
+2. When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+
+3. The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+
+4. After both players have rolled and chosen dice order, the player with the higher combined number wins.
+*/
+
+// Global variables
+var currentMode = "Roll the dice";
+var myOutputValue = "";
+var player = "Player 1";
+var player1Results = "";
+var player2Results = "";
+var player1Score = 0;
+var player2Score = 0;
+
+// Generate dices
+var diceRollGenerator = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomDiceRoll = Math.floor(randomDecimal) + 1;
+  return randomDiceRoll;
+};
+
+// Specifying the 2 dices
+var diceRoll1 = diceRollGenerator();
+var diceRoll2 = diceRollGenerator();
```

Comment:
These dice should be declared as global variables but with empty strings. They have already been reassigned in lines 43 and 44. We do not want to assign them to diceRollGenerator() here as the values will be fixed and will not change for player 2. However inside of the function at lines 43-44, every time generateBeatThat() runs, diceRollGenerator() will execute again and you will get a new value for diceRoll1 and diceRoll2.